src/main/java/org/semux/core/Block.java

```diff
@@ -400,158 +397,13 @@ public long getTimestamp() {
-     *
-     * @return
-     */
-    public byte[] getEncodedVotes() {
-        SimpleEncoder enc = new SimpleEncoder(4 + 4 + votes.size() * Signature.LENGTH);
-
-        enc.writeInt(view);
-        enc.writeInt(votes.size());
-        for (Signature vote : votes) {
-            enc.writeBytes(vote.toBytes());
-        }
-
-        return enc.toBytes();
-    }
-
-    /**
-     * Parses a block instance from bytes.
-     *
-     * @param h
-     *            Serialized header
-     * @param t
-     *            Serialized transactions
-     * @param r
-     *            Serialized transaction results
-     * @param v
-     *            Serialized votes
-     * @return
-     */
-    public static Block fromComponents(byte[] h, byte[] t, byte[] r, byte[] v) {
-        if (h == null) {
-            throw new IllegalArgumentException("Block header can't be null");
-        }
-        if (t == null) {
-            throw new IllegalArgumentException("Block transactions can't be null");
-        }
-
-        BlockHeader header = BlockHeader.fromBytes(h);
-
-        SimpleDecoder dec = new SimpleDecoder(t);
-        List<Transaction> transactions = new ArrayList<>();
-        int n = dec.readInt();
-        for (int i = 0; i < n; i++) {
-            transactions.add(Transaction.fromBytes(dec.readBytes()));
-        }
-
-        List<TransactionResult> results = new ArrayList<>();
-        if (r != null) {
-            dec = new SimpleDecoder(r);
-            n = dec.readInt();
-            for (int i = 0; i < n; i++) {
-                results.add(TransactionResult.fromBytes(dec.readBytes()));
-            }
-        }
-
-        int view = 0;
-        List<Signature> votes = new ArrayList<>();
-        if (v != null) {
-            dec = new SimpleDecoder(v);
-            view = dec.readInt();
-            n = dec.readInt();
-            for (int i = 0; i < n; i++) {
-                votes.add(Signature.fromBytes(dec.readBytes()));
-            }
-        }
-
-        return new Block(header, transactions, results, view, votes);
-    }
-
-    public byte[] toBytes() {
-        SimpleEncoder enc = new SimpleEncoder();
-        enc.writeBytes(getEncodedHeader());
-        enc.writeBytes(getEncodedTransactions());
-        enc.writeBytes(getEncodedResults());
-        enc.writeBytes(getEncodedVotes());
-
-        return enc.toBytes();
-    }
-
-    public static Block fromBytes(byte[] bytes) {
-        SimpleDecoder dec = new SimpleDecoder(bytes);
-        byte[] header = dec.readBytes();
-        byte[] transactions = dec.readBytes();
-        byte[] results = dec.readBytes();
-        byte[] votes = dec.readBytes();
-
-        return Block.fromComponents(header, transactions, results, votes);
-    }
-
     /**
      * Get block size in bytes
      *
      * @return block size in bytes
      */
     public int size() {
-        return toBytes().length;
+        return new BlockEncoderV1().encode(this).length;
```

Comment:
Will this always be v1 block?

Reply:
We should propose a new block limit, considering that we're also enabling EVM.